examples/ui_actions_explorer/public/actions/actions.tsx

```diff
@@ -34,32 +34,32 @@ export const EDIT_USER_ACTION = 'EDIT_USER_ACTION';
 export const PHONE_USER_ACTION = 'PHONE_USER_ACTION';
 export const SHOWCASE_PLUGGABILITY_ACTION = 'SHOWCASE_PLUGGABILITY_ACTION';
 
-export const showcasePluggability = createAction({
+export const showcasePluggability = createAction<typeof SHOWCASE_PLUGGABILITY_ACTION>({
   type: SHOWCASE_PLUGGABILITY_ACTION,
   getDisplayName: () => 'This is pluggable! Any plugin can inject their actions here.',
   execute: async () => alert("Isn't that cool?!"),
 });
 
 export type PhoneContext = string;
 
-export const makePhoneCallAction = createAction<PhoneContext>({
+export const makePhoneCallAction = createAction<typeof CALL_PHONE_NUMBER_ACTION>({
   type: CALL_PHONE_NUMBER_ACTION,
   getDisplayName: () => 'Call phone number',
   execute: async phone => alert(`Pretend calling ${phone}...`),
 });
 
-export const lookUpWeatherAction = createAction<{ country: string }>({
+export const lookUpWeatherAction = createAction<typeof TRAVEL_GUIDE_ACTION>({
   type: TRAVEL_GUIDE_ACTION,
   getIconType: () => 'popout',
   getDisplayName: () => 'View travel guide',
-  execute: async ({ country }) => {
+  execute: async country => {
     window.open(`https://www.worldtravelguide.net/?s=${country},`, '_blank');
   },
 });
 
 export type CountryContext = string;
```

Comment:
I think will be better in this example to make all contexts into objects. And in general suggest it as best practice. 

Right now from typescript perspective CountryContext and PhoneContext are the same: `string`. So we technically allowed to use country context where phone context is expected and other way around. 

If we make it into {country: string} and {phone: string} it would be better and typescript won't allow to use one where other is expected.

Reply:
Yea, that's a good point. Plus makes it much easier to extend the interface without making breaking changes. 

I can require it. It was originally required to be an object.  What do you think?  Require via typescript, or just encourage?

Reply:
If it is not a big trouble to require it via typescript, I'd go for it!

Reply:
Turned out to help anyway with the ts issue you spotted, so, done!